### ISSUE TITLE
Angular 1.5 Route

### ISSUE THREAD

**Issue by veenayp** [angular/angular#6803]
Won't we have nested route option?

if we place 'ng-outlet' inside the component as a place holder for nested control, browser hangs.

**Comment by brandonroberts** [comment]
@veenayp can you reproduce in a plunker?

**Comment by veenayp** [comment]
http://plnkr.co/edit/vsGG0OBATjpanZSIAsMG?p=preview

In preview section, click home or dashboard, component based on route will be loaded.

Dashboard component has multiple components... 
FYI: In side dashboard components are loaded dynamically based on entitlement.

Particularly for pipeline component, loaded inside dashboard, i want to create nested route.
refer to pipeline.js, after uncommenting line #10, placed it into line # 7. template will look like below...
And now when you run, browser will hang :(
FYI: nested ng-outlet is included to load component based on nested route...

template: '<div class="col-sm-4"> \
    	<h3>Pipeline</h3> \
		<p>New customer looking to lend there secuitites.</p> \
		<p> <a class="btn btn-lg btn-outline" >View More</a> </p> \
		<ng-outlet></ng-outlet>\
	</div>'

Hope this explains my problem.

Thanks for the support!

**Comment by brandonroberts** [comment]
This could be a bug, but I think the way you are trying to use ng-outlet is causing an infinite loop. Nested outlets do work though: See modified example: http://plnkr.co/edit/1LWH4MHphQRwDsZXuiqs?p=preview

**Comment by brandonroberts** [comment]
See also: http://stackoverflow.com/questions/28914706/angular-compile-in-directive-seems-to-go-into-infinite-loop

**Comment by veenayp** [comment]
@brandonroberts seems you have deleted previous plunker: http://plnkr.co/edit/1LWH4MHphQRwDsZXuiqs?p=preview. where you suggested way around for nested route...

Could you please provide the same?

**Comment by brandonroberts** [comment]
@veenayp sure, here you go: http://plnkr.co/edit/qYLy7Jlfl7BOlAzc3wIG?p=preview

**Comment by veenayp** [comment]
Thanks! This is very helpful

What your suggestion for modal view as child route? Which could be open
from any route.
For example: if user click profile link, a overlay window should appear
with portfolio component. User could click portfolio from dashboard or home.

Route could be:
\home\portfolio
Or
\dashboard \portfolio
Or
\dashboard\message\inbox\portfolio

How to append child route dynamically to a current navigated route?

**Comment by brandonroberts** [comment]
I think what you are looking for are auxiliary routes, which can be navigated to in parallel to regular routes. These type of routes aren't supported in the Angular 1 router build yet though.

**Comment by veenayp** [comment]
ohk :(

you means once we have defined routeConfig, we can't append more child route...

for example, once have provided rout config in component, as below
$routeConfig: [
    { path: '/child1', component: 'child1', name: 'Child1' },
    { path: '/child2', component: 'child2', name: 'Child2' },
    { path: '/child3', component: 'child3', name: 'Child3' }
]

later, at run time, we can't add child route child4?
    { path: '/child4', component: 'child4', name: 'Child4' }

**Comment by brandonroberts** [comment]
Yes, you can dynamically add child routes. You'd have to inject the router and add them that way. You can't load multiple routes at once yet though.

**Comment by veenayp** [comment]
Thanks brandon for quick response :)

referring below plunker, could you please help me in adding 3 dynamic child route.
Plunker: http://plnkr.co/edit/smTI3GUJpjbOAVbCG7VC?p=preview


1. { path: '/level3', component: 'level3', name: 'Level3' }

rout info comes from service: (home.js, service.getCommonRout())
2. { path: '/help', component: 'help', name: 'Help' }
3. { path: '/profile', component: 'profile', name: 'Profile' }

**Comment by brandonroberts** [comment]
Here is a working fork of adding dynamic routes. http://plnkr.co/edit/MHErayGMUaK61In95KPq?p=preview

Your dynamic routes won't work on direct navigation though because they won't be defined when the navigation is triggered, but that may be a separate issue. I suggest you define your routes upfront and dynamically load your components instead.
